### PR TITLE
npm: check that package-lock.json gets updated

### PR DESCRIPTION
**Pull request by robin-da**
This adds a test to see that package-lock.json gets updated if a package gets
added in the generated js dependencies.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: language-support/ts/codegen/tests/BUILD.bazel [hunk 1]
```diff
@@ -114,6 +114,23 @@ sh_test(
     ],
 ) if not is_windows else None
 
+sh_test(
+    name="npm-update-check",
+    srcs = ["npm_check.sh"],
```
Comment (stefanobaghino-da) [inline language-support/ts/codegen/tests/BUILD.bazel:119]: Should this test be skipped on Windows?

File: language-support/ts/codegen/tests/npm_check.sh [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+# --- begin runfiles.bash initialization v2 ---
+# Copy-pasted from the Bazel Bash runfiles library v2.
+    set -uo pipefail; f=bazel_tools/tools/bash/runfiles/runfiles.bash
+    source "${RUNFILES_DIR:-/dev/null}/$f" 2>/dev/null || \
+      source "$(grep -sm1 "^$f " "${RUNFILES_MANIFEST_FILE:-/dev/null}" | cut -f2- -d' ')" 2>/dev/null || \
+      source "$0.runfiles/$f" 2>/dev/null || \
+      source "$(grep -sm1 "^$f " "$0.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
+      source "$(grep -sm1 "^$f " "$0.exe.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
+      { echo>&2 "ERROR: cannot find $f"; exit 1; }; f=; set -e
```
Comment (stefanobaghino-da) [inline language-support/ts/codegen/tests/npm_check.sh:18]: Probably worth removing the duplicated `set`s at the beginning and end and fixing the indentation.

**Review by stefanobaghino-da — APPROVED** [review]
💚

File: language-support/ts/codegen/tests/BUILD.bazel (hunk 1 above)
Comment (cocreature) [reply language-support/ts/codegen/tests/BUILD.bazel:119]: unfortunately yes

File: language-support/ts/codegen/tests/npm_check.sh (hunk 2 above)
Comment (cocreature) [reply language-support/ts/codegen/tests/npm_check.sh:18]: this is copy-pasted (you can’t avoid it, this is a bootstrapping problem, this is the logic for locating dependencies, if you try to factor it out, you can’t locate the factored out code). I wouldn’t modify this in any way.

File: language-support/ts/codegen/tests/npm_check.sh [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      with
+        controller p
+          do return (p, p)
+EOF
+
+rm npmcheck.dar
+$DAML build -o npmcheck.dar
+$DAML codegen js npmcheck.dar -o "$TMP_DIR"/daml.js
+$NPM install
+PACKAGELOCK2=$(md5sum "$TMP_DIR"/package-lock.json | awk '{print $1}')
```
Comment (cocreature) [inline language-support/ts/codegen/tests/npm_check.sh:97]: Can we throw `jq` at this to get the actual dependencies and verify that the list got longer by 1?

**Review by cocreature — APPROVED** [review]
Nice, thank you!

File: language-support/ts/codegen/tests/npm_check.sh (hunk 2 above)
Comment (stefanobaghino-da) [reply language-support/ts/codegen/tests/npm_check.sh:18]: I hoped that removing the `set`s and changing the indentation was safe enough, but I trust your judgement. 🙂

File: language-support/ts/codegen/tests/BUILD.bazel (hunk 1 above)
Comment (robin-da) [reply language-support/ts/codegen/tests/BUILD.bazel:119]: disabled it for windows.

File: language-support/ts/codegen/tests/npm_check.sh (hunk 3 above)
Comment (robin-da) [reply language-support/ts/codegen/tests/npm_check.sh]: using jq now.

**Pull request merged**